packages/components/organisms/f-checkout/src/components/Header.vue

```diff
@@ -0,0 +1,182 @@
+<template>
+    <div
+        data-test-id="checkoutHeader-component"
+        :class="$style['c-checkoutHeader']">
+        <div
+            v-if="isLoggedIn"
+            data-test-id="user-checkoutHeader">
+            <h2 data-test-id="user-title">
+                {{ $t('checkoutHeader.user.title', { name }) }}
+            </h2>
+
+            <p :class="$style['c-checkoutHeader-loginLink']">
+                <a
+                    :href="loginUrl"
+                    data-test-id="switch-user-link"
+                    @click="onVisitLoginPage">
+                    {{ $t('checkoutHeader.user.switchUser', { name }) }}
+                </a>
+            </p>
+        </div>
+
+        <div
+            v-else
+            data-test-id="guest-checkoutHeader">
+            <h2 :class="$style['c-checkoutHeader-title']">
+                {{ $t('checkoutHeader.guest.guestTitle') }}
+            </h2>
+
+            <a
+                :href="loginUrl"
+                data-test-id="guest-login-button"
+                :class="$style['c-checkoutHeader-loginButton']"
+                @click="onVisitLoginPage">
+                {{ $t('checkoutHeader.guest.loginButton') }}
+            </a>
+
+            <div
+                :class="$style['c-checkoutHeader-option']">
+                <span>{{ $t('checkoutHeader.guest.option') }}</span>
+            </div>
+
+            <h2 :class="$style['c-checkoutHeader-title']">
+                {{ $t('checkoutHeader.guest.guestTitle') }}
+            </h2>
+
+            <p :class="$style['c-checkoutHeader-confirmation']">
+                {{ $t('checkoutHeader.guest.confirmation') }}
+            </p>
+        </div>
+    </div>
+</template>
+
+<script>
+import { mapState } from 'vuex';
+import EventNames from '../event-names';
+
+export default {
+    props: {
+        loginUrl: {
+            type: String,
+            required: true
+        }
+    },
+
+    computed: {
+        ...mapState('checkout', [
+            'customer',
+            'isLoggedIn'
+        ]),
+
+        name () {
+            return (this.customer.firstName.charAt(0).toUpperCase() + this.customer.firstName.slice(1));
+        }
+    },
+
+    methods: {
+        onVisitLoginPage () {
```

Comment:
JSDoc here would be helpful, something along the lines of why this is being emitted and what's listening to it/what happens after that event is triggered.

Reply:
will update in follow up PR 🙂